ucm2/sof-soundwire/sof-soundwire.conf

```diff
@@ -4,3 +4,196 @@ SectionUseCase."HiFi" {
 	File "HiFi.conf"
 	Comment "Play HiFi quality Music"
 }
+
+# the kcontrols initial values, which will be set by `alsactl init`
+
+If.pga1p {
+	Condition {
+		Type ControlExists
+		Control "name='PGA1.0 1 Master Playback Volume'"
```

Comment:
this is unmaintainable. No one will ever remember what those controls are now that you have moved them between files. I really would like each init value to be located in the file for which this value makes sense. Putting everything together goes against the modularity we wanted to add here.

Reply:
Unfortunately, it's the global section for the first boot hardware initialization which is not tied with the verb/device contents. I am not against to put the relevant configs to the sub-configs (codecs/ tree or so), but if you have everything together, just put it to the main config file. We can modify it later.

Reply:
There is no way we can guarantee thant PGA1.0.1 Master Playback Volume maps to a RT1308 amplifier or a Maxim 98373 amplifier. The premise is that we would do a generic/modular driver, it doesn't seem right to me to put the init values together when they are meant to be independent.

Reply:
Then follow the main condition block in verb HiFi:

````
If.RT700 {
	Condition {
		Type String
		Haystack "${CardComponents}"
        	Needle "hs:rt700" 
	}
	True {
                BootSequence [
                     ....
                ]
        }
}
````

I'm ok to put this to RT700-init.conf and inlude those files from sof-soundwire.conf to follow the verb structure.

Reply:
> There is no way we can guarantee thant PGA1.0.1 Master Playback Volume maps to a RT1308 amplifier or a Maxim 98373 amplifier.

@plbossart As there is no guarantee that PGA1 or PGA2 will map to which codec,  we should put PGA settings in a general file. Otherwise, it may introduce some problems. For example, PGA1 maybe used by RT711 on some platforms, but on other platforms, PGA2 is used by RT711.

@plbossart @perexg I think this is a good idea that we can move codec specific settings into the codec specific config, for example, moving the RT700 settings into RT700-init.conf. We may need update alsa-lib to support such usage. Currently alsa-lib only support BootSequence in the card config file.

Reply:
> @plbossart @perexg I think this is a good idea that we can move codec specific settings into the codec specific config, for example, moving the RT700 settings into RT700-init.conf. We may need update alsa-lib to support such usage. Currently alsa-lib only support BootSequence in the card config file.

And it's correct. It's the global config. Just add <sof-soundwire/RT700-init.conf> lines to the card config file (sof-soundwire.conf).

Reply:
> > There is no way we can guarantee thant PGA1.0.1 Master Playback Volume maps to a RT1308 amplifier or a Maxim 98373 amplifier.
> 
> @plbossart As there is no guarantee that PGA1 or PGA2 will map to which codec, we should put PGA settings in a general file. Otherwise, it may introduce some problems. For example, PGA1 maybe used by RT711 on some platforms, but on other platforms, PGA2 is used by RT711.

PGA settings for the topology should always be 100%/0dB by default. There is no reason why we should do any attenuation in the firmware, it's much better to do such attenuations in the codec.

> @plbossart @perexg I think this is a good idea that we can move codec specific settings into the codec specific config, for example, moving the RT700 settings into RT700-init.conf. We may need update alsa-lib to support such usage. Currently alsa-lib only support BootSequence in the card config file.

That would be fine and aligned with what we did for previous solutions, we could even put the codec configurations in the codec/ directory as we did for legacy platforms.

Reply:
> That would be fine and aligned with what we did for previous solutions, we could even put the codec configurations in the codec/ directory as we did for legacy platforms.

The PGA (dsp) related configs should be in dsp/ directory. It's a dsp config, not the codec config.

Reply:
> PGA settings for the topology should always be 100%/0dB by default. There is no reason why we should do any attenuation in the firmware, it's much better to do such attenuations in the codec.

@plbossart For playback pipeline PGA, 32 is 0dB. The original UCM setting such as `cset "name='PGA3.0 3 Master Playback Volume' 50"` is wrong and '50' is out of the range. Please see the example of PGA1:
```
numid=54,iface=MIXER,name='PGA1.0 1 Master Playback Volume'
  ; type=INTEGER,access=rw---R--,values=2,min=0,max=32,step=0
  : values=32,32
  | dBscale-min=-64.00dB,step=2.00dB,mute=1
```

@plbossart @perexg If we align that we should put PGA setting in dsp/ directory and codec setting in codec/ directory, I will update the patch.